### PR TITLE
fix(#275): implement weekly log rotation

### PR DESCRIPTION
**Pull request by Phil-Venter**
Added weekly log rotation.

This fix creates a `RotatingFileHandler` that extends the `Mono/RotatingFileHandler` and overwrites only the methods required to make this work. Might be considered hacky by some.

### CODE REVIEW

File: src/Tempest/Log/src/FileHandlers/RotatingFileHandler.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        }
+
+        $this->dateFormat = $dateFormat;
+    }
+
+    protected function getNextRotation(): DateTimeImmutable
+    {
+        return match (str_replace(['/','_','.'], '-', $this->dateFormat)) {
+            self::FILE_PER_WEEK => (new DateTimeImmutable('first day of next week'))->setTime(0, 0, 0),
+            self::FILE_PER_MONTH => (new DateTimeImmutable('first day of next month'))->setTime(0, 0, 0),
```
Comment (brendt) [inline src/Tempest/Log/src/FileHandlers/RotatingFileHandler.php:33]: Where are these other constants?

File: src/Tempest/Log/src/FileHandlers/RotatingFileHandler.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+declare(strict_types=1);
+
+namespace Tempest\Log\FileHandlers;
+
+use DateTimeImmutable;
+use InvalidArgumentException;
+use Monolog\Handler\RotatingFileHandler as MonoRotatingFileHandler;
+
+final class RotatingFileHandler extends MonoRotatingFileHandler
```
Comment (brendt) [inline src/Tempest/Log/src/FileHandlers/RotatingFileHandler.php:11]: I think this could be refactored to an enum, wouldn't that be cleaner?

**Review by brendt — CHANGES_REQUESTED** [review]

File: src/Tempest/Log/src/FileHandlers/RotatingFileHandler.php (hunk 1 above)
Comment (Phil-Venter) [reply src/Tempest/Log/src/FileHandlers/RotatingFileHandler.php:33]: They are defined at the top of the base `Monolog\Handler\RotatingFileHandler`

I wanted to create a minimal invasive "wrapper" over `Monolog\Handler\RotatingFileHandler` that just adds support for weekly

image

File: src/Tempest/Log/src/FileHandlers/RotatingFileHandler.php (hunk 2 above)
Comment (Phil-Venter) [reply src/Tempest/Log/src/FileHandlers/RotatingFileHandler.php:11]: You want to convert `RotatingFileHandler::FILE_PER_WEEK` etc to enums... I can do this, are you sure we want to introduce that much changes to essentially internal monolog functionality, if it ever changes more will have to be updated in that case. Or am I misunderstanding something??

Comment (brendt) [reply src/Tempest/Log/src/FileHandlers/RotatingFileHandler.php:11]: Oh I totally missed the extend part 😂 No I get it now :)

File: src/Tempest/Log/src/FileHandlers/RotatingFileHandler.php (hunk 1 above)
Comment (brendt) [reply src/Tempest/Log/src/FileHandlers/RotatingFileHandler.php:33]: Yeah, totally missed that this class was extending monolog 😁

**Review by brendt — APPROVED** [review]

**Comment by brendt** [comment]
Sorry for missing the extension part :) Thanks for contributing!

**Pull request merged**